### PR TITLE
feat(protocol-designer): add DOM identifiers for e2e tests

### LINKED ISSUE #7607: PD: Identifiers needed for e2e tests - Advanced settings Transfer Form
Below elements needs the identifier to work on the advanced settings of Transfer form.

Batch Edit mode:

- [x] 1. Tip Position, the field itself without clicking on the input box -- so I can verify the `_ mm` number of tip position value **[IN PROGRESS by @shlokamin]**

Added an id `TipPositionField_FIELDNAME` where FIELDNAME is the name of the field. So for mix tip position, it would be `TipPositionField_mix_mmFromBottom`
- [x] 2. Selector for Tip Position Modal -> radio buttons, and the field -- so I can enter the number I want, and so I can toggle btw custom and default radio buttons **[IN PROGRESS by @shlokamin]**

Added a name inside the whole radio group that contains both buttons. It can be identified via `name='TipPositionOptions'`
Once you get the DOM element, you can then query for an `input` field with `value` of either `default` or `custom`

- [x] 3. Selectors for the + and - buttons in Tip Position Modal -- so I can click them and make sure the custom field value changes **[IN PROGRESS by @shlokamin]**

Added two ids to the buttons: `<button id="Increment_tipPosition" />` and `<button id="Decrement_tipPosition" />` 

- [x] 4. Well Order (Aspirate and Dispense) -- so I can identify the well order. If it's "mixed" (indeterminate state), I want to see that. (Idea: `data-test="t2b,l2r"` or similar) **[IN PROGRESS by @shlokamin]**

Added ids to the buttons to open the well order modal: `id="WellOrderField_button"`

Once the modal is opened, the dropdown options can be queried via semantic HTML as is: 

```
<select name="aspirate_wellOrder_first"">
   <option value="t2b">Top to bottom</option>
   <option value="b2t">Bottom to top</option>
   <option value="l2r">Left to right</option>
   <option value="r2l">Right to left</option>
</select>
```

- [x] 5. Delay tip position fields (aspirate/dispense/touch tip, all inputs with tooltip 'distance from the bottom of the well') need a selector so we can read the value of mm

See number 2 above, can be queried using the new id: 
```
<input id="TipPositionField_aspirate_delay_mmFromBottom" type="text" readonly="" value="22">
``` 

- [x] 6. The path buttons need selectors so they can be clicked. Also need to be able to tell if they're selected/deselected/disabled (they're not button elements, need some way to clearly see that they are selected/deselected/disabled)

Added an id so the path buttons can be clicked, can be accessed via 

`id="PathButton_PATH_SELECTED"`, where PATH is the path, and SELECTED is `selected` if selected, and `deselected` otherwise

i.e.:

```
<li id="PathButton_single_selected"></li>
<li id="PathButton_multiAspirate_deselected"></li>
<li id="PathButton_multiDispense_deselected"></li>
```

- [x] 7. TBD: Blowout location dropdown - need to figure out (source well, destination well and Trash) -- I want to read and change the values of Blowout location

Added id="BlowoutLocationField_dropdown" to the dropdown.

i.e.
```
<select  id="BlowoutLocationField_dropdown">
   <option value="trashId">Trash</option>
   <option value="source_well">Source Well</option>
   <option value="dest_well">Destination Well</option>
</select>
```

- [x] 8. Pipette dropdown has no `name`?

added id="PipetteField_dropdown"

i.e.

```
<select " id="PipetteField_dropdown">
   <option value="56ac2870-9bae-11eb-8a4b-e7b378696bf6">P20 Single-Channel GEN2</option>
   <option value="56ac2871-9bae-11eb-8a4b-e7b378696bf6">P20 8-Channel GEN2</option>
</select>
```

NOTE: all/most of the dropdowns seem to not have a `name` -- double check what's going on

LATER: deck state selectors, once we figure out what we need to test in Deck State

### PR DESCRIPTION
**Pull request by shlokamin**
# Overview

This PR adds a bunch of identifiers to be used by e2e tests.

closes #7607

# Changelog
- Add DOM identifiers for e2e tests

# Review requests
Code review, make sure names and all that make sense
Make sure identifiers show up on the DOM

# Risk assessment
Low

### CODE REVIEW

File: protocol-designer/src/components/StepEditForm/fields/PathField/Path.js [hunk 1]
```diff
@@ -89,13 +99,17 @@ const getSubtitle = (
   return reasonForDisabled || ''
 }
 
+const getPathButtonId = (name, selectedValue) =>
+  `PathButton_${name}_${name === selectedValue ? 'selected' : 'deselected'}`
+
```
Comment (shlokamin) [inline protocol-designer/src/components/StepEditForm/fields/PathField/Path.js]: don't love having whether the path option is selected or deselected in the id, but not sure of a better way to do this

File: protocol-designer/src/components/StepEditForm/fields/WellOrderField/index.js [hunk 2]
```diff
@@ -109,9 +109,14 @@ export const WellOrderField = (props: Props): React.Node => {
...
               )}
+              id="WellOrderField_button"
             />
           ) : (
-            <Text onClick={handleOpen} css={mixedWellOrderStyles}>
+            <Text
+              onClick={handleOpen}
+              css={mixedWellOrderStyles}
+              id="WellOrderField_button"
+            >
```
Comment (shlokamin) [inline protocol-designer/src/components/StepEditForm/fields/WellOrderField/index.js]: these two won't ever show up on the DOM at the same time, so its cool to use the same id i think. alternatively, i could do 

`id="WellOrderField_button"` and `id="WellOrderField_button_mixed"` or something

**Review by IanLondon — CHANGES_REQUESTED** [review]
Missing: within the tip position modal, if you click "custom", the `input type="text"` field where you can enter `mm` still could use a selector.

✔️ Good move: `name=TipPositionOptions` appears on two `input` elements. And that's correct, for radio button `input` elements! I didn't realize radio buttons in a group are supposed to share `name`, I learned something new :D

---

For well order field: I think Neha wanted to be able to "read" the button without opening the modal (which way the zig-zag icon is oriented. In the ticket, we had the idea `data-test="t2b,l2r"` to represent first: top to bottom & second: left to right, for example. to do right to left, top to bottom the `data-test` of the icon/button would be `data-test="r2l,t2b"` and so on)

Also, ids are supposed to be unique but there are 2x `id=WellOrderField_button` elements in Transfer. Common id for aspirate well order vs dispense well order.  Maybe address by doing `WellOrderField_button_${aspirate|dispense}` or something like that?

---

For selector number 6, the path button selectors, I think they're supposed to differentiate disabled vs simply deselected. Right now you can only tell that they're deselected, but I'm pretty sure we want to be able to confirm that they're disabled or not. It might be easier for E2E if the ID is constant for each button (eg `id="PathButton_single"` with no modifiers to the `id`) and `data-test` can be the lookup for selected vs deselected, and disabled or not disabled??

---

For selector number 7, the `disposal volume > blowout` field may need a selector?

---

For selector number 8, would it make more sense for it to be `name="pipette"` rather than using `id="PipetteField_dropdown"` to match eg the `name="aspirate_labware"` and `name="dispense_labware"` dropdowns?

**Comment by shlokamin** [comment]
Great feedback, thanks @IanLondon. I like this PR much better now 😄 

> Missing: within the tip position modal, if you click "custom", the `input type="text"` field where you can enter `mm` still could use a selector.

Good catch, added `<input type="text" value="" id="TipPositionModal_custom-input">`

> For well order field: I think Neha wanted to be able to "read" the button without opening the modal (which way the zig-zag icon is oriented. In the ticket, we had the idea `data-test="t2b,l2r"` to represent first: top to bottom & second: left to right, for example. to do right to left, top to bottom the `data-test` of the icon/button would be `data-test="r2l,t2b"` and so on)
> 
> Also, ids are supposed to be unique but there are 2x `id=WellOrderField_button` elements in Transfer. Common id for aspirate well order vs dispense well order. Maybe address by doing `WellOrderField_button_${aspirate|dispense}` or something like that?

Gocha, I added data-test but also kept the id so we can more semantically click on the button if need be. Also, to follow our convention of `${ComponentName}_${Indentifer}` where component name is pascal case identifier is snake case I did:


```js
<img id="WellOrderField_button_aspirate" data-test="WellOrderField_button_t2b_l2r">
```


> For selector number 6, the path button selectors, I think they're supposed to differentiate disabled vs simply deselected. Right now you can only tell that they're deselected, but I'm pretty sure we want to be able to confirm that they're disabled or not. It might be easier for E2E if the ID is constant for each button (eg `id="PathButton_single"` with no modifiers to the `id`) and `data-test` can be the lookup for selected vs deselected, and disabled or not disabled??

Oof, this one kinda sucks because we're adding a lot of extra stuff here. I wish we could more semantically query the class name (since thats is actually what is being used to render what the button actually means), but that is not very reliable. Anyway, I did:

```js
<li id="PathButton_single" data-test="PathButton_selected_enabled"></li>
```



> For selector number 7, the `disposal volume > blowout` field may need a selector?

Good call, I added:

```js
<select id="DisposalVolumeField_dropdown">
   <option value="trashId">Trash</option>
   <option value="source_well">Source Well</option>
   <option value="dest_well" disabled="">Destination Well</option>
</select>
```

> For selector number 8, would it make more sense for it to be `name="pipette"` rather than using `id="PipetteField_dropdown"` to match eg the `name="aspirate_labware"` and `name="dispense_labware"` dropdowns?

Another good call! I did:

```js
<select name="pipette">
   <option value="56ac2870-9bae-11eb-8a4b-e7b378696bf6">P20 Single-Channel GEN2</option>
   <option value="56ac2871-9bae-11eb-8a4b-e7b378696bf6">P20 8-Channel GEN2</option>
</select>
```

**Review by IanLondon — APPROVED** [review]
changes look good!!

**Review by Kadee80 — APPROVED** [review]
🔚 🥈 🔚

**Pull request merged**